posthog/hogql/printer.py

```diff
@@ -199,95 +301,208 @@ def print_ast(
                 )
 
             # check that we're not running inside another aggregate
-            for stack_node in stack:
+            for stack_node in self.stack:
                 if stack_node != node and isinstance(stack_node, ast.Call) and stack_node.name in HOGQL_AGGREGATIONS:
                     raise ValueError(
                         f"Aggregation '{node.name}' cannot be nested inside another aggregation '{stack_node.name}'."
                     )
 
-            translated_args = ", ".join([print_ast(arg, stack, context, dialect) for arg in node.args])
-            if dialect == "hogql":
-                response = f"{node.name}({translated_args})"
+            translated_args = ", ".join([self.visit(arg) for arg in node.args])
+            if self.dialect == "hogql":
+                return f"{node.name}({translated_args})"
             elif node.name == "count":
-                response = "count(*)"
+                return "count(*)"
```

Comment:
https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference/count/ - count() is optimised by clickhouse
```suggestion
                return "count()"
```

Reply:
This will be fixed together with the distinct count later on :).